hardhat.config.ts

```diff
@@ -38,9 +40,24 @@ if (["rinkeby", "mainnet"].includes(argv.network) && INFURA_KEY === undefined) {
   );
 }
 
+const mocha: MochaOptions = {};
+switch (MOCHA_CONF) {
+  case undefined:
+    break;
+  case "coverage":
+    mocha.grep = /^(?!E2E)/;
```

Comment:
Maybe add a short comment explaining why we need to ignore E2E tests for code coverage? While it was clear in the description, I think it would also be nice to keep it here so we don't forget when looking at the configuration file.